internal/jimm/juju/jimm.go

```diff
@@ -287,7 +287,20 @@ func (j *JujuManager) PrepareModelMigration(
 			return err
 		}
 
-		if err := j.Database.AddIncomingModelMigration(ctx, &dbmodel.IncomingModelMigration{
+		// Verify the model doesn't exist - if it does that means a migration is
+		// in progress or completed or it could also mean the model failed to be removed
+		// during migration ABORT but that problem should be dealt with separately.
+		model := &dbmodel.Model{
+			UUID: sql.NullString{String: modelUUID, Valid: true},
+		}
+		err := j.Database.GetModel(ctx, model)
```

Comment:
Could make a little func wrapping this called:

"CanPrepareModelMigration" or something, then you can put this doc there. Cause really it's just checking that it's safe to Prepare a model migration and can be godoc'd on a func.